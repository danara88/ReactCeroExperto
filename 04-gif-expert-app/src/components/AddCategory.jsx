import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {
  //const inputRef = useRef();

  // Hooks
  const [inputValue, setInputValue] = useState('');

  // Methods
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    // setCategories((categories) => [...categories, inputRef.current.value]);
    // setCategories((categories) => [inputValue, ...categories]);

    // Emit value to parent component
    onNewCategory(inputValue.trim());

    setInputValue('');
  };

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        // ref={inputRef}
        // onChange={(event) => onInputChange(event)}
        onChange={onInputChange}
      ></input>
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
