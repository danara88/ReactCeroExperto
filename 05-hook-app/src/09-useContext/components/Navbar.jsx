import { Link, NavLink } from 'react-router-dom';

export const NavBar = () => {
  return (
    <>
      <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <div className="container-fluid">
          <Link to="/" className="navbar-brand" href="#">
            UseContext
          </Link>
          <button
            className="navbar-toggler"
            type="button"
            data-bs-toggle="collapse"
            data-bs-target="#navbarNavAltMarkup"
            aria-controls="navbarNavAltMarkup"
            aria-expanded="false"
            aria-label="Toggle navigation"
          >
            <span className="navbar-toggler-icon"></span>
          </button>
          <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
            <div className="navbar-nav">
              <NavLink
                to="/home"
                className={({ isActive }) => `nav-link ${isActive ? 'active' : ''}`}
              >
                Home
              </NavLink>
              <NavLink
                to="/about"
                className={({ isActive }) => `nav-link ${isActive ? 'active' : ''}`}
              >
                About
              </NavLink>
              <NavLink
                to="/login"
                className={({ isActive }) => `nav-link ${isActive ? 'active' : ''}`}
              >
                Login
              </NavLink>
            </div>
          </div>
        </div>
      </nav>
    </>
  );
};
