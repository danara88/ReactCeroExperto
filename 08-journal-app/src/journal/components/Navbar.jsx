import { LogoutOutlined, MenuOutlined } from '@mui/icons-material';
import { AppBar, Grid2, IconButton, Toolbar, Typography } from '@mui/material';

/**
 * @constant Navbar
 * @description Navbar component
 */
export const Navbar = ({ drawerWidth = 240 }) => {
  return (
    <AppBar
      position="fixed"
      sx={{ width: { sm: `calc(100% - ${drawerWidth}px)` }, ml: { sm: `${drawerWidth}px` } }}
    >
      <Toolbar>
        <IconButton color="inherit" edge="start" sx={{ mr: 2, display: { sm: 'none' } }}>
          <MenuOutlined />
        </IconButton>

        <Grid2
          container
          direction="row"
          justifyContent="space-between"
          alignItems="center"
          width="100%"
        >
          <Typography variant="h6" noWrap component="div">
            Journal App
          </Typography>
          <IconButton color="error">
            <LogoutOutlined />
          </IconButton>
        </Grid2>
      </Toolbar>
    </AppBar>
  );
};
