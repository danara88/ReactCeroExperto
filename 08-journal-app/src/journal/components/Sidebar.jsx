import { TurnedInNot } from '@mui/icons-material';
import {
  Box,
  Divider,
  Drawer,
  Grid2,
  List,
  ListItem,
  ListItemButton,
  ListItemIcon,
  ListItemText,
  Toolbar,
  Typography,
} from '@mui/material';

/**
 * @constant Sidebar
 * @description Sidebar component
 */
export const Siderbar = ({ drawerWidth = 240 }) => {
  return (
    <Box component="nav" sx={{ width: { sm: drawerWidth }, flexShrink: { sm: 0 } }}>
      <Drawer
        variant="permanent"
        open
        sx={{
          display: { sx: 'block' },
          '& .MuiDrawer-paper': { boxSizing: 'border-box', width: drawerWidth },
        }}
      >
        <Toolbar>
          <Typography variant="h6">Daniel Aranda</Typography>
        </Toolbar>
        <Divider />

        <List>
          {['Enero', 'Febrero', 'Marzo'].map((text) => (
            <ListItem key={text} disablePadding>
              <ListItemButton>
                <ListItemIcon>
                  <TurnedInNot />
                </ListItemIcon>
                <Grid2 container>
                  <ListItemText primary={text} />
                  <ListItemText secondary={'Hello word this a fixed text'} />
                </Grid2>
              </ListItemButton>
            </ListItem>
          ))}
        </List>
      </Drawer>
    </Box>
  );
};
