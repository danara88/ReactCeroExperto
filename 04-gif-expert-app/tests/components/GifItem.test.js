import { describe, expect, test } from '@jest/globals';
import { render, screen } from '@testing-library/react';
import { GifItem } from '../../src/components/GifItem';

describe('GifItem', () => {
  const title = 'Test';
  const url = 'http://test.com/';

  test('should evaluate snap shot', () => {
    // Act
    const { container } = render(<GifItem title={title} url={url} />);
    // Assert
    expect(container).toMatchSnapshot();
  });

  test('should display correct image form URL and correct title', () => {
    // Arrange
    render(<GifItem title={title} url={url} />);
    // Act
    const { src, alt } = screen.getByRole('img');
    // Assert
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test('should display the title in the UI', () => {
    // Arrange
    render(<GifItem title={title} url={url} />);
    // Act
    const result = screen.getByText(title);
    // Assert
    expect(result).toBeTruthy();
  });
});
