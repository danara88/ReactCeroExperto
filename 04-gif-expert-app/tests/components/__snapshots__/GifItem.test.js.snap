// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem should evaluate snap shot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Test"
      src="http://test.com/"
    />
    <p>
      Test
    </p>
  </div>
</div>
`;
